r3bgen/R3BParticleGenerator.h

```diff
@@ -0,0 +1,103 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#ifndef R3BGENERATOR_H
+#define R3BGENERATOR_H
+
+#include "FairGenerator.h"
+#include "R3BDistribution.h"
+#include "R3BDistribution1D.h"
+#include "TRandom3.h"
+
+#include <iostream>
+
+class FairPrimaryGenerator;
+
+// C++17 will make the following lines much better, but for now we have to stick with this...
+// maybe not very good to read, but its at least constexpr
+constexpr int ChargeMass2PDGCode(int z, uint a)
+{
+    return (z > 0 ? (1000000000 + z * 10000 + a * 10) : -(1000000000 - z * 10000 + a * 10));
+}
+constexpr int PDGCode2Mass(int pdg)
+{
+    return (pdg > 1000000000 ? ((pdg / 10) % 1000) : (pdg < 1000000000 ? ((-pdg / 10) % 1000) : 0));
+}
+constexpr int PDGCode2Charge(int pdg)
+{
+    return (pdg > 1000000000 ? ((pdg / 10000) % 1000) : (pdg < 1000000000 ? -((-pdg / 10000) % 1000) : 0));
+}
+constexpr std::array<int, 2> PDGCode2ChargeMass(int pdg) { return { PDGCode2Charge(pdg), PDGCode2Mass(pdg) }; }
+
+class R3BParticleGenerator : public FairGenerator
+{
+
+  public:
+    R3BParticleGenerator();
+
+    R3BParticleGenerator(Int_t pdgCode, UInt_t seed = 0);
+    R3BParticleGenerator(const Char_t* ionName, unsigned int seed = 0);
+
+    void SetProton();
+    void SetNeutron();
+    void SetIon(const Char_t* ionName);
+    void SetParticle(Int_t pdgCode);
+    void SetParticle(Int_t charge, UInt_t mass)
```

Comment:
Instead of charge / mass rather Z, A to avoid confusion with mass in GeV/c².

Note that there is a huge overlap to the other generators, e.g. [`R3BPhaseSpaceGenerator`](https://github.com/R3BRootGroup/R3BRoot/blob/8affc0f5150f0b8927f5c11a66cf29ceba0fed42/r3bgen/R3BPhaseSpaceGenerator.cxx#L50) That might be something to extract and de-duplicate.